doc/development/create-a-model.md

```diff
@@ -0,0 +1,71 @@
+# Create a model
+
+If you'd like to create a new model that isn't covered by the existing DeePMD-kit library, but reuse DeePMD-kit's other efficient module such as data processing, trainner, etc, you may want to read this section.
+
+To incorporate your custom model you'll need to:
+1. Register and implement new components (e.g. descriptor) in a Python file. You may also want to regiester new TensorFlow OPs if necessary.
+2. Register new arguments for user inputs.
+3. Package new codes into a Python package.
+4. Test new models.
+
+## Design a new component
+
+When creating a new component, take descriptor as the example, you should inherit [`Descriptor`](deepmd.descriptor.descriptor.Descriptor) class and override several methods. Abstract methods such as [`build`](deepmd.descriptor.descriptor.Descriptor.build) must be implemented and others are not. You should keep arguments of these methods unchanged.
```

Comment:
The links are broken in markdown

Reply:
The links will be parsed in sphinx. I have no idea how to make it also work in GitHub, so I change to {py:class}`deepmd.descriptor.descriptor.Descriptor` which will also generate links in sphinx.